File: lab1/fp_SimUDuck/RubberDuck.cpp
#include "stdafx.h"
#include "RubberDuck.h"

RubberDuck::RubberDuck()
	:Duck(FlyBehaviors::GetFlyNoWay(),
		QuackBehaviors::GetSqueakBehavior(),
		DanceBehaviors::GetNoDanceBehavior())
{
}

void RubberDuck::Display() const
{
	std::cout << "I'm rubber duck" << std::endl;
}

File: lw1_fn/src/duck/RubberDuck.php
<?php

class RubberDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(
            FlyBehavior\getFlyNoWay(),
            QuackBehavior\getSqueakBehavior(),
            DanceBehavior\getNoDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m rubber duck!' . PHP_EOL;
    }
}